x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -44,9 +44,46 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *types.MsgStakeSupplie
 	if !isSupplierFound {
 		logger.Info(fmt.Sprintf("Supplier not found. Creating new supplier for address %q", msg.Address))
 		supplier = k.createSupplier(ctx, msg)
+
+		// Ensure that only the owner can stake a new supplier.
+		if err = supplier.EnsureOwner(msg.Sender); err != nil {
+			logger.Error(fmt.Sprintf(
+				"owner address %q in the message does not match the msg sender address %q",
+				msg.OwnerAddress,
+				msg.Sender,
+			))
+
+			return nil, err
```

Comment:
Can you add a custom error here that captures "Ensure that only the owner can stake a new supplier."

Reply:
To mimic Morse behavior, now both can stake a new supplier.